### ISSUE TITLE
Many addons don't work if dynamically enabled

### ISSUE THREAD

**Issue by mxmou** [ScratchAddons/ScratchAddons#6136]
### Description

Dynamically enabling any addon that uses `addon.tab.scratchClass` in a userscript with `"runAtComplete": true` results in the following error:
```
Uncaught (in promise) Error: Wait until addon.tab.scratchClassReady() resolves before using addon.tab.scratchClass
    at Tab.scratchClass (Tab.js:289:13)
```

The following addons are affected:
* better-img-uploads
* color-picker (only in code editor)
* echo-effect
* editor-messages
* find-bar
* gamepad
* hide-stage
* swap-local-global
* variable-manager

### Steps to reproduce

1. Reload the editor
2. Enable one of the addons listed above
3. Try to use the addon

### Environment

Browser: Edge 115 dev
Scratch Addons version: f9813518
Operating system: Windows

**Comment by BroJac5246** [comment]
Do the addons still work despite the error (I can't test this right now)?

**Comment by WorldLanguages** [comment]
> Seems like a regression of #6105. cc @WorldLanguages

Yeah, looks like a test case I missed. Fix should be easy though.
Nice catch mxmou!

**Comment by pufferfish101007** [comment]
This does not only occur when dynamically enabling addons; it seems that if another `"runAtComplete": true` runs first and calls `vm.emitWorkspaceUpdate()` (at least I think that's what's causing it) then it also causes this behaviour - presumably this will also be fixed by #6137
